src/Th_List.ml

```diff
@@ -0,0 +1,22 @@
+open Core
+open Components
+open Types
+
+let vlist = VList []
+
+let new_components = [
+  {
+    name = "length";
+    codomain = TInt;
+    domain = [TList];
+    check = (function
+             | _ -> true
+            );
+    apply = (function
+             | [VList x] -> VInt (List.length x)
+             | _ -> VError);
+    dump = (fun [@warning "-8"] [a] -> "(len " ^ a ^ ")")
+  } ;
+]
+
+let all_components =  new_components
```

Comment:
1. Please add the remaining List components from old PIE repository.
2. For now, export `Th_LIA.components` here. Using `Th_List` and choosing between LIA vs NIA would need a small change on my side. Please leave a comment that we use `Th_List` with only `Th_LIA` for now.